rotkehlchen/rotkehlchen.py

```diff
@@ -537,13 +534,24 @@ def _logout(self) -> None:
         # Make sure no messages leak to other user sessions
         self.msg_aggregator.consume_errors()
         self.msg_aggregator.consume_warnings()
+        self.task_manager.clear()  # type: ignore  # task_manage is not None here
         self.task_manager = None
 
+        # release the locks in the globaldb. We saw that when killing
+        # a greenlet the locks are not released. We have locks in the
+        # chain aggregator that gets removed in this function and the
+        # db connections. The user db gets replaced but the globaldb
+        # doesn't so just in case we release the locks. It won't raise
+        # errors if the lock is over-released
+        # https://www.gevent.org/api/gevent.lock.html#gevent.lock.Semaphore.release
+        # The killall that happens in this logic can trigger a greenlet switch as per
+        # https://github.com/gevent/gevent/issues/1473#issuecomment-548327614
+        GlobalDBHandler().packaged_db_lock.release()
```

Comment:
Abstract it into a function of `clear_locks()` of GlobalDB. Move the comment to the function's docstring. And call it here